x/ethbridge/keeper/keeper.go

```diff
@@ -95,32 +96,91 @@ func (k Keeper) ProcessSuccessfulClaim(ctx sdk.Context, claim string, sugaredLog
 }
 
 // ProcessBurn processes the burn of bridged coins from the given sender
-func (k Keeper) ProcessBurn(ctx sdk.Context, cosmosSender sdk.AccAddress, amount sdk.Coins, sugaredLogger *zap.SugaredLogger) error {
-	if err := k.supplyKeeper.SendCoinsFromAccountToModule(
-		ctx, cosmosSender, types.ModuleName, amount,
-	); err != nil {
-		sugaredLogger.Errorw("failed to process burn.",
+func (k Keeper) ProcessBurn(ctx sdk.Context, cosmosSender sdk.AccAddress, msg types.MsgBurn, sugaredLogger *zap.SugaredLogger) error {
+	var coins sdk.Coins
+
+	if msg.Symbol == types.CethSymbol {
+		coins = sdk.NewCoins(sdk.NewCoin(types.CethSymbol, msg.CethAmount.Add(msg.Amount)))
+	} else {
+		coins = sdk.NewCoins(sdk.NewCoin(msg.Symbol, msg.Amount), sdk.NewCoin(types.CethSymbol, msg.CethAmount))
+	}
+
+	err := k.supplyKeeper.SendCoinsFromAccountToModule(
+		ctx, cosmosSender, types.ModuleName, coins,
+	)
+
+	if err != nil {
+		sugaredLogger.Errorw("failed to send coin from account to module.",
 			errorMessageKey, err.Error())
 		return err
 	}
 
-	if err := k.supplyKeeper.BurnCoins(ctx, types.ModuleName, amount); err != nil {
-		panic(err)
+	coins = sdk.NewCoins(sdk.NewCoin(types.CethSymbol, msg.CethAmount))
+	err = k.supplyKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, k.GetCethReceiverAccount(ctx), coins)
+	if err != nil {
+		sugaredLogger.Errorw("failed to send ceth from module to account.",
+			errorMessageKey, err.Error())
+		return err
+	}
+
+	coins = sdk.NewCoins(sdk.NewCoin(msg.Symbol, msg.Amount))
+	err = k.supplyKeeper.BurnCoins(ctx, types.ModuleName, coins)
+	if err != nil {
+		sugaredLogger.Errorw("failed to burn locked coin.",
+			errorMessageKey, err.Error())
+		return err
 	}
 
 	return nil
 }
 
 // ProcessLock processes the lockup of cosmos coins from the given sender
-func (k Keeper) ProcessLock(ctx sdk.Context, cosmosSender sdk.AccAddress, amount sdk.Coins) error {
-	return k.supplyKeeper.SendCoinsFromAccountToModule(ctx, cosmosSender, types.ModuleName, amount)
+func (k Keeper) ProcessLock(ctx sdk.Context, cosmosSender sdk.AccAddress, msg types.MsgLock, sugaredLogger *zap.SugaredLogger) error {
+	coins := sdk.NewCoins(sdk.NewCoin(msg.Symbol, msg.Amount), sdk.NewCoin(types.CethSymbol, msg.CethAmount))
+
+	err := k.supplyKeeper.SendCoinsFromAccountToModule(ctx, cosmosSender, types.ModuleName, coins)
```

Comment:
Are we sending coins from a user to the module, and then afterwards sending that ceth to the ceth receiver account?

Reply:
Didn't find the solution to make a transaction including both burn the locked token and transfer ceth. so send both coins to module at first.

Reply:
ProcessLock should be atomic and it is unnecessary to implement like this way. will change it.

Reply:
no interface from account to account transfer in supplyKeeper.